torch/_inductor/mkldnn_lowerings.py

```diff
@@ -612,25 +633,25 @@ def qlinear_unary(
             bias_dtype = None if bias is None else bias.get_dtype()
 
             choices: List[ChoiceCaller] = []
+
             if use_max_autotune():
                 *_, layout, x, packed_weight = mm_args(
                     x, packed_weight, layout=layout, out_dtype=output_dtype
                 )
-                if (
-                    isinstance(
-                        ir.InputsKernel.unwrap_storage_for_input(x_zp),
-                        ir.ConstantBuffer,
-                    )
-                    and len(x_zp.get_layout().size) == 0  # Per tensor quant of act
-                    and isinstance(
+
+                def _is_sym_quantized_wgt():
+                    # We don't support the GEMM template with asymmetrically quantized weights.
+                    nonlocal w_zp
```

Comment:
Why do we need this?

Reply:
Oops, sorry! I didn't notice your comment & started merging. Will try canceling it

Reply:
This check was already present but @leslie-fang-intel advised creating a separate function would make the code more readable.

Reply:
> This check was already present but @leslie-fang-intel advised creating a separate function would make the code more readable.

The suggestion is based on your previous implementation, which included too many redundant changes. After all these refactors, the current implementation is now simple and readable enough without requiring a separate function.

Reply:
> The suggestion is based on your previous implementation, which included too many redundant changes. After all these refactors, 

Can you please point out which change was redundant? 

The current code in the main branch assumes zero point would be a 0D tensor, so it is not covering the case in which zero-point is 1D, as in some test_cpp_wrapper.py cases.

Reply:
@jgong @leslie-fang-intel, I now refactored the check on `w_zp` to decrease the diff with the main branch, but I retained the new check on `x_zp` outside the check for GEMM template use feasibility because it's also applicable for oneDNN qlinear.

Thanks!

.